pkg/staking/stakingcontract/contract.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stakingcontract
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethersphere/bee/pkg/sctx"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bee/pkg/transaction"
+	"github.com/ethersphere/go-sw3-abi/sw3abi"
+)
+
+var (
+	MinimumStakeAmount = big.NewInt(1)
+
+	erc20ABI = parseABI(sw3abi.ERC20ABIv0_3_1)
+	//TODO: get ABI for staking contract and replace it below
+	stakingABI = parseABI(sw3abi.ERC20ABIv0_3_1)
+	//TODO: enable below mentioned topic for receiving receipts
+	//stakeUpdatedTopic = stakingABI.Events["StakeUpdated"].ID
+
+	ErrInsufficentStakeAmount = errors.New("insufficient stake amount")
+	ErrInsufficientFunds      = errors.New("insufficient token balance")
+	ErrNotImplemented         = errors.New("not implemented")
+	ErrGetStakeFailed         = errors.New("get stake failed")
+
+	depositStakeDescription = "Deposit Stake"
+)
+
+type Contract interface {
+	DepositStake(ctx context.Context, stakedAmount big.Int, overlay swarm.Address) error
+	GetStake(ctx context.Context, overlay swarm.Address) (big.Int, error)
```

Comment:
Not sure why the `big.Int` in this signature is not a pointer. You get it back from geth as a pointer, then you return it as object, and then in the API you use the reference again. Instead, this could just be `*big.Int` and we can avoid two type conversions.

Reply:
it was a suggestion by @vladopajic to avoid using pointers if the value isn't supposed to change.

Reply:
Hmm.... Not sure if this makes any sense. Even if it's a pointer or object, once it's returned the caller is the custodian. It doesn't matter either way.

It makes sense to return the pointer in this case as we save 2 type castings.

Reply:
Pass by value should be preferred way so any changes made inside function does not affect original value. [playground example](https://go.dev/play/p/KI4zhru4S6c) 

What I haven't had in mind is how internals of `big.Int` are implemented - which can't make it immutable if value copied.

> 2 type castings
 
 (kindly drawing attention) getting value from pointer is dereference operation

Reply:
@swarmHaseeb let pass the pointer, let's not make this a blocker

Reply:
> Pass by value should be preferred way so any changes made inside function does not affect original value. [playground example](https://go.dev/play/p/KI4zhru4S6c)

So I was talking about the return value of the `GetStake` function. The returned pointer will never change. Passing it in the DepositStake as a value seems better I agree.

> (kindly drawing attention) getting value from pointer is dereference operation

Yes sorry my bad, dereferencing. Either way, I would prefer not to do unnecessary dereferencing. Having the Deposit call signature with value is okay, but in the API code, the return value of GetStake can be used as is.